### DISCUSSION TITLE
UV mapping from primitives?

### DISCUSSION THREAD

**Discussion by NodeGuy** [elalish/manifold#417] [General]
I'm new to all this so please be gentle with me.

I understand it's possible to generate complex shapes by starting with primitives such as **cube** and **sphere** and combining them with boolean operations.

Is it similarly possible to generate complex UV mappings by starting with primitive shapes and combining them with boolean operations so that they can, e.g., be fed to Three.js for rendering textures?

**Comment by elalish** [comment]
So, yes and no. First, our primitives don't come with UVs, so you'll have to make them yourself (or import them from three.js). Second, when you boolean shapes together, we do keep track of the UVs for you, but they're separate. If you want to atlas your materials into a single set of textures, you'll need to update those UVs to shift them around next to each other. We don't have an automatic function for that either. 

However, if you want to load multiple objects with their own textures and then join the objects and keep using those same materials as separate draw calls, that works pretty much out of the box.

**Reply by NodeGuy** [reply]
OK, thank you and thank you for this wonderful library!
